queries/haskell.scm

```diff
@@ -0,0 +1,268 @@
+; argumentOrParameter
+; "arg" function parameter or function call argument
+(function
+  .
+  (variable) @functionName
+  .
+  (patterns
+    (
+      (_) @argumentOrParameter
+    ) @_.iteration
```

Comment:
I'm guessing you don't want this to be the iteration scope for `functionName`, so you probably want
```suggestion
    ) @argumentOrParameter.iteration
```

Reply:
wait, so the iteration scopes aren't scoped?

Reply:
that would be nice, but no. not worth the complexity. fwiw we don't recommend defining iteration scope at the same time as scope, because it actually generates one copy of the iteration scope per scope, which we then silently merge